src/sentry/integrations/msteams/utils.py

```diff
@@ -105,6 +105,8 @@ def send_incident_alert_notification(
     from .card_builder import build_incident_attachment
 
     attachment = build_incident_attachment(incident, new_status, metric_value)
+    if action.target_identifier is None:
+        raise ValueError("Can't send without `target_identifier`")
```

Comment:
This error condition shouldn't be hit with any existing paths, as it would have led to an error later anyway.